Realm/RLMRealm.mm

```diff
@@ -712,6 +713,40 @@ + (BOOL)performMigrationForConfiguration:(RLMRealmConfiguration *)configuration
     return success;
 }
 
+- (RLMThreadHandover *)exportThreadHandoverWithObjects:(NSArray<id<RLMThreadConfined>> *)objects {
+    return [[RLMThreadHandover alloc] initWithRealm:self objects:objects];
+}
+
+- (void)transactionAsyncWithBlock:(void(^)(RLMRealm *))block {
+    [self transactionAsyncWithObjects:@[] block:^(RLMRealm * _Nonnull realm,
+                                                  NSArray<id<RLMThreadConfined>> * _Nonnull objects) {
+        REALM_ASSERT_DEBUG(objects.count == 0);
+        block(realm);
+    }];
+}
+
+- (void)transactionAsyncWithObject:(id<RLMThreadConfined>)object
+                             block:(void(^)(RLMRealm *, id<RLMThreadConfined>))block {
+    [self transactionAsyncWithObjects:@[object] block:^(RLMRealm * _Nonnull realm,
+                                                        NSArray<id<RLMThreadConfined>> * _Nonnull objects) {
+        REALM_ASSERT_DEBUG(objects.count == 1);
+        block(realm, objects[0]);
+    }];
+}
+
+- (void)transactionAsyncWithObjects:(NSArray<id<RLMThreadConfined>> *)objects
+                              block:(void(^)(RLMRealm *, NSArray<id<RLMThreadConfined>> *))block {
+    RLMThreadHandover *package = [self exportThreadHandoverWithObjects:objects];
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0), ^{
```

Comment:
I'm not sure about using `DISPATCH_QUEUE_PRIORITY_BACKGROUND` here. Nothing about an async write inherently indicates that it's of a lower priority. For instance, a priority of user initiated may make more sense if the write is triggered by a user action and the app is expecting the write to trigger an update of the UI. The background quality of service class throttles I/O, which may result in the transaction commit taking significantly longer than if it were running at a higher QoS class. The background priority queues are also serviced after higher-priority queues, meaning they can be starved if the app is performing a lot of work on any of the higher-priority queues.